Build Failures

### What Happened?

Build failure while trying to build the version from git

### Steps to Reproduce

Download the git?

### Expected Behavior

Successful build from git?

### OS Version

Other Linux

### Software Version

Compiled from git

### Log Output

```shell
329/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkListBox.c.o
FAILED: src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkListBox.c.o 
cc -Isrc/io.elementary.files.p -Isrc -I../pantheon-files/src -Ilibcore -I../pantheon-files/libcore -I. -I../pantheon-files -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/gee-0.8 -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/cloudproviders -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/fribidi -I/usr/include/granite -I/usr/include/libhandy-1 -I/usr/include/zeitgeist-2.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch '-DGETTEXT_PACKAGE="io.elementary.files"' -DHANDY_USE_UNSTABLE_API -w -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/pantheon-files-git/src=/usr/src/debug/pantheon-files-git -flto=auto -fPIE -pthread -D_REENTRANT -MD -MQ src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkListBox.c.o -MF src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkListBox.c.o.d -o src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkListBox.c.o -c src/io.elementary.files.p/View/Sidebar/BookmarkListBox.c
src/io.elementary.files.p/View/Sidebar/BookmarkListBox.c: In function ‘sidebar_bookmark_list_box_constructor’:
src/io.elementary.files.p/View/Sidebar/BookmarkListBox.c:1198:9: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_list_box’; did you mean ‘sidebar_sidebar_list_interface_get_list_box’? [-Wimplicit-function-declaration]
 1198 |         sidebar_sidebar_list_interface_set_list_box ((SidebarSidebarListInterface*) self, _tmp1_);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         sidebar_sidebar_list_interface_get_list_box
src/io.elementary.files.p/View/Sidebar/BookmarkListBox.c: In function ‘_vala_sidebar_bookmark_list_box_set_property’:
src/io.elementary.files.p/View/Sidebar/BookmarkListBox.c:1329:17: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_sidebar’; did you mean ‘sidebar_sidebar_list_interface_get_sidebar’? [-Wimplicit-function-declaration]
 1329 |                 sidebar_sidebar_list_interface_set_sidebar ((SidebarSidebarListInterface*) self, g_value_get_object (value));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 sidebar_sidebar_list_interface_get_sidebar
[330/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Window.c.o
[331/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkRow.c.o
FAILED: src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkRow.c.o 
cc -Isrc/io.elementary.files.p -Isrc -I../pantheon-files/src -Ilibcore -I../pantheon-files/libcore -I. -I../pantheon-files -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/gee-0.8 -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/cloudproviders -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/fribidi -I/usr/include/granite -I/usr/include/libhandy-1 -I/usr/include/zeitgeist-2.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch '-DGETTEXT_PACKAGE="io.elementary.files"' -DHANDY_USE_UNSTABLE_API -w -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/pantheon-files-git/src=/usr/src/debug/pantheon-files-git -flto=auto -fPIE -pthread -D_REENTRANT -MD -MQ src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkRow.c.o -MF src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkRow.c.o.d -o src/io.elementary.files.p/meson-generated_View_Sidebar_BookmarkRow.c.o -c src/io.elementary.files.p/View/Sidebar/BookmarkRow.c
src/io.elementary.files.p/View/Sidebar/BookmarkRow.c: In function ‘sidebar_bookmark_row_constructor’:
src/io.elementary.files.p/View/Sidebar/BookmarkRow.c:2312:9: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_id’; did you mean ‘sidebar_sidebar_item_interface_get_id’? [-Wimplicit-function-declaration]
 2312 |         sidebar_sidebar_item_interface_set_id ((SidebarSidebarItemInterface*) self, sidebar_sidebar_item_interface_get_next_item_id ());
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         sidebar_sidebar_item_interface_get_id
src/io.elementary.files.p/View/Sidebar/BookmarkRow.c: In function ‘_vala_sidebar_bookmark_row_set_property’:
src/io.elementary.files.p/View/Sidebar/BookmarkRow.c:2666:17: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_list’; did you mean ‘sidebar_sidebar_item_interface_get_list’? [-Wimplicit-function-declaration]
 2666 |                 sidebar_sidebar_item_interface_set_list ((SidebarSidebarItemInterface*) self, g_value_get_object (value));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 sidebar_sidebar_item_interface_get_list
src/io.elementary.files.p/View/Sidebar/BookmarkRow.c:2678:17: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_pinned’; did you mean ‘sidebar_sidebar_item_interface_get_pinned’? [-Wimplicit-function-declaration]
 2678 |                 sidebar_sidebar_item_interface_set_pinned ((SidebarSidebarItemInterface*) self, g_value_get_boolean (value));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 sidebar_sidebar_item_interface_get_pinned
src/io.elementary.files.p/View/Sidebar/BookmarkRow.c:2681:17: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_permanent’; did you mean ‘sidebar_sidebar_item_interface_get_permanent’? [-Wimplicit-function-declaration]
 2681 |                 sidebar_sidebar_item_interface_set_permanent ((SidebarSidebarItemInterface*) self, g_value_get_boolean (value));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 sidebar_sidebar_item_interface_get_permanent
[332/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_DeviceListBox.c.o
FAILED: src/io.elementary.files.p/meson-generated_View_Sidebar_DeviceListBox.c.o 
cc -Isrc/io.elementary.files.p -Isrc -I../pantheon-files/src -Ilibcore -I../pantheon-files/libcore -I. -I../pantheon-files -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/gee-0.8 -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/cloudproviders -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/fribidi -I/usr/include/granite -I/usr/include/libhandy-1 -I/usr/include/zeitgeist-2.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch '-DGETTEXT_PACKAGE="io.elementary.files"' -DHANDY_USE_UNSTABLE_API -w -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/pantheon-files-git/src=/usr/src/debug/pantheon-files-git -flto=auto -fPIE -pthread -D_REENTRANT -MD -MQ src/io.elementary.files.p/meson-generated_View_Sidebar_DeviceListBox.c.o -MF src/io.elementary.files.p/meson-generated_View_Sidebar_DeviceListBox.c.o.d -o src/io.elementary.files.p/meson-generated_View_Sidebar_DeviceListBox.c.o -c src/io.elementary.files.p/View/Sidebar/DeviceListBox.c
src/io.elementary.files.p/View/Sidebar/DeviceListBox.c: In function ‘sidebar_device_list_box_constructor’:
src/io.elementary.files.p/View/Sidebar/DeviceListBox.c:1246:9: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_list_box’; did you mean ‘sidebar_sidebar_list_interface_get_list_box’? [-Wimplicit-function-declaration]
 1246 |         sidebar_sidebar_list_interface_set_list_box ((SidebarSidebarListInterface*) self, _tmp1_);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         sidebar_sidebar_list_interface_get_list_box
src/io.elementary.files.p/View/Sidebar/DeviceListBox.c: In function ‘_vala_sidebar_device_list_box_set_property’:
src/io.elementary.files.p/View/Sidebar/DeviceListBox.c:1375:17: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_sidebar’; did you mean ‘sidebar_sidebar_list_interface_get_sidebar’? [-Wimplicit-function-declaration]
 1375 |                 sidebar_sidebar_list_interface_set_sidebar ((SidebarSidebarListInterface*) self, g_value_get_object (value));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 sidebar_sidebar_list_interface_get_sidebar
[333/354] Compiling C object src/io.elementary.files.p/meson-generated_View_AbstractDirectoryView.c.o
[334/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_AbstractMountableRow.c.o
[335/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_NetworkListBox.c.o
FAILED: src/io.elementary.files.p/meson-generated_View_Sidebar_NetworkListBox.c.o 
cc -Isrc/io.elementary.files.p -Isrc -I../pantheon-files/src -Ilibcore -I../pantheon-files/libcore -I. -I../pantheon-files -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/gee-0.8 -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/cloudproviders -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/fribidi -I/usr/include/granite -I/usr/include/libhandy-1 -I/usr/include/zeitgeist-2.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch '-DGETTEXT_PACKAGE="io.elementary.files"' -DHANDY_USE_UNSTABLE_API -w -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/pantheon-files-git/src=/usr/src/debug/pantheon-files-git -flto=auto -fPIE -pthread -D_REENTRANT -MD -MQ src/io.elementary.files.p/meson-generated_View_Sidebar_NetworkListBox.c.o -MF src/io.elementary.files.p/meson-generated_View_Sidebar_NetworkListBox.c.o.d -o src/io.elementary.files.p/meson-generated_View_Sidebar_NetworkListBox.c.o -c src/io.elementary.files.p/View/Sidebar/NetworkListBox.c
src/io.elementary.files.p/View/Sidebar/NetworkListBox.c: In function ‘sidebar_network_list_box_constructor’:
src/io.elementary.files.p/View/Sidebar/NetworkListBox.c:734:9: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_list_box’; did you mean ‘sidebar_sidebar_list_interface_get_list_box’? [-Wimplicit-function-declaration]
  734 |         sidebar_sidebar_list_interface_set_list_box ((SidebarSidebarListInterface*) self, _tmp1_);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         sidebar_sidebar_list_interface_get_list_box
src/io.elementary.files.p/View/Sidebar/NetworkListBox.c: In function ‘_vala_sidebar_network_list_box_set_property’:
src/io.elementary.files.p/View/Sidebar/NetworkListBox.c:857:17: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_sidebar’; did you mean ‘sidebar_sidebar_list_interface_get_sidebar’? [-Wimplicit-function-declaration]
  857 |                 sidebar_sidebar_list_interface_set_sidebar ((SidebarSidebarListInterface*) self, g_value_get_object (value));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 sidebar_sidebar_list_interface_get_sidebar
[336/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_DriveRow.c.o
[337/354] Linking target plugins/pantheon-files-cloud/libpantheon-files-cloudproviders.so
[338/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_NetworkRow.c.o
[339/354] Linking target plugins/pantheon-files-trash/libpantheon-files-trash.so
[340/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Widgets_SearchResults.c.o
[341/354] Linking target plugins/contractor/libpantheon-files-contractor.so
[342/354] Linking target libcore/tests/GOFFileTests/GOFFileTests
[343/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_SidebarWindow.c.o
[344/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_VolumeRow.c.o
[345/354] Linking target libcore/tests/MarlinIconInfoTests/MarlinIconInfoTests
[346/354] Compiling C object src/io.elementary.files.p/meson-generated_View_Sidebar_VolumelessMountRow.c.o
[347/354] Linking target libcore/tests/GOFDirectoryAsyncTests/GOFDirectoryAsyncTests
[348/354] Linking target libcore/tests/UtilTests/UtilTests
[349/354] Linking target plugins/pantheon-files-ctags/libpantheon-files-ctags.so
[350/354] Linking target pantheon-files-daemon/io.elementary.files-daemon
ninja: build stopped: subcommand failed.
```


### Hardware Info

_No response_

--- a/filechooser-portal/Main.vala
+++ b/filechooser-portal/Main.vala
@@ -511,7 +511,6 @@ public class Files.FileChooserPortal : Object {
 
         var loop = new MainLoop (null, false);
         try {
-            var session_bus = Bus.get_sync (BusType.SESSION);
             var owner_id = Bus.own_name (
                 BusType.SESSION,
                 "org.freedesktop.impl.portal.desktop.elementary.files",

--- a/filechooser-portal/Main.vala
+++ b/filechooser-portal/Main.vala
@@ -478,6 +478,7 @@ public class Files.FileChooserPortal : Object {
             connection.register_object ("/org/freedesktop/portal/desktop", new FileChooserPortal (connection));
         } catch (Error e) {
             critical ("Unable to register the object: %s", e.message);
+            // Should we quit main loop if this happens?
         }
     }
 
@@ -510,21 +511,16 @@ public class Files.FileChooserPortal : Object {
         }
 
         var loop = new MainLoop (null, false);
-        try {
-            var owner_id = Bus.own_name (
-                BusType.SESSION,
-                "org.freedesktop.impl.portal.desktop.elementary.files",
-                BusNameOwnerFlags.ALLOW_REPLACEMENT | (opt_replace ? BusNameOwnerFlags.REPLACE : 0),
-                on_bus_acquired,
-                () => debug ("org.freedesktop.impl.portal.desktop.elementary.files acquired"),
-                () => loop.quit ()
-            );
-            loop.run ();
-            Bus.unown_name (owner_id);
-        } catch (Error e) {
-            printerr ("No session bus: %s\n", e.message);
-            return 2;
-        }
+        var owner_id = Bus.own_name (
+            BusType.SESSION,
+            "org.freedesktop.impl.portal.desktop.elementary.files",
+            BusNameOwnerFlags.ALLOW_REPLACEMENT | (opt_replace ? BusNameOwnerFlags.REPLACE : 0),
+            on_bus_acquired,
+            () => debug ("org.freedesktop.impl.portal.desktop.elementary.files acquired"),
+            () => loop.quit ()
+        );
+        loop.run ();
+        Bus.unown_name (owner_id);
 
         return 0;
     }

--- a/libcore/IconInfo.vala
+++ b/libcore/IconInfo.vala
@@ -213,10 +213,6 @@ public class Files.IconInfo : GLib.Object {
             scale = _scale;
         }
 
-        public LoadableIconKey dup () {
-            return new LoadableIconKey (icon, size, scale);
-        }
-
         public static uint hash (LoadableIconKey a) {
             return a.icon.hash () ^ a.size;
         }
@@ -236,17 +232,6 @@ public class Files.IconInfo : GLib.Object {
             icon = _icon;
             size = _size;
             scale = _scale;
-        }
-
-        public uint hash () {
-            return icon.hash () ^ (size * scale);
-        }
-
-        public bool equal (ThemedIconKey other) {
-            if (this.size != other.size || this.scale != other.scale)
-                return false;
-
-            return this.icon.equal (other.icon);
         }
     }
 

--- a/libcore/Interfaces/SidebarItemInterface.vala
+++ b/libcore/Interfaces/SidebarItemInterface.vala
@@ -42,13 +42,13 @@ public interface Sidebar.SidebarItemInterface : Object {
         return item;
     }
 
-    public abstract SidebarListInterface list { get; construct; }
-    public abstract uint32 id { get; construct; }
+    public abstract SidebarListInterface list { get; set construct; }
+    public abstract uint32 id { get; set construct; }
     public abstract string uri { get; set construct; }
     public abstract string custom_name { get; set ; }
     public abstract Icon gicon { get; set construct; }
-    public abstract bool pinned { get; construct; } // Whether can be dragged
-    public abstract bool permanent { get; construct; } // Whether can be deleted
+    public abstract bool pinned { get; set construct; } // Whether can be dragged
+    public abstract bool permanent { get; set construct; } // Whether can be deleted
     public abstract bool can_insert_before { get; set; default = true; }
     public abstract bool can_insert_after { get; set; default = true; }
     public abstract int get_index ();

--- a/libcore/Interfaces/SidebarListInterface.vala
+++ b/libcore/Interfaces/SidebarListInterface.vala
@@ -6,8 +6,8 @@
  */
 
 public interface Sidebar.SidebarListInterface : Object {
-    public abstract Files.SidebarInterface sidebar { get; construct; }
-    public abstract Gtk.ListBox list_box { get; internal set; }
+    public abstract Files.SidebarInterface sidebar { get; set construct; }
+    public abstract Gtk.ListBox list_box { get; set construct; }
 
     public abstract void select_item (Gtk.ListBoxRow? item);
     public abstract void unselect_all_items ();

--- a/plugins/pantheon-files-ctags/plugin.vala
+++ b/plugins/pantheon-files-ctags/plugin.vala
@@ -108,7 +108,7 @@ public class Files.Plugins.CTags : Files.Plugins.Base {
         menu_item.show ();
     }
 
-    private async void set_color (GLib.List<Files.File> files, int n) throws IOError {
+    private async void set_color (GLib.List<Files.File> files, int n) throws Error {
         foreach (unowned Files.File file in files) {
             if (!(file is Files.File)) {
                 continue;

--- a/src/View/Sidebar/BookmarkListBox.vala
+++ b/src/View/Sidebar/BookmarkListBox.vala
@@ -21,8 +21,8 @@
  */
 
 public class Sidebar.BookmarkListBox : Gtk.Box, Sidebar.SidebarListInterface {
-    public Files.SidebarInterface sidebar { get; construct; }
-    public Gtk.ListBox list_box { get; internal set; }
+    public Files.SidebarInterface sidebar { get; set construct; }
+    public Gtk.ListBox list_box { get; set construct; }
 
     private Files.BookmarkList bookmark_list;
     private unowned Files.TrashMonitor trash_monitor;

--- a/src/View/Sidebar/BookmarkRow.vala
+++ b/src/View/Sidebar/BookmarkRow.vala
@@ -41,12 +41,12 @@ public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
 
     /* Each row gets a unique id.  The methods relating to this are in the SidebarItemInterface */
     static construct {
-        SidebarItemInterface.row_id = new Rand.with_seed (
+        row_id = new Rand.with_seed (
             int.parse (get_real_time ().to_string ())
         ).next_int ();
 
-        SidebarItemInterface.item_map_lock = Mutex ();
-        SidebarItemInterface.item_id_map = new Gee.HashMap<uint32, SidebarItemInterface> ();
+        item_map_lock = Mutex ();
+        item_id_map = new Gee.HashMap<uint32, SidebarItemInterface> ();
     }
 
     private bool valid = true; //Set to false if scheduled for removal
@@ -117,9 +117,9 @@ public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
         set_tooltip_text (Files.FileUtils.sanitize_path (uri, null, false));
 
         selectable = true;
-        id = SidebarItemInterface.get_next_item_id ();
+        id = get_next_item_id ();
         item_map_lock.@lock ();
-        SidebarItemInterface.item_id_map.@set (id, this);
+        item_id_map.@set (id, this);
         item_map_lock.unlock ();
 
         label = new Gtk.Label (display_name) {
@@ -219,7 +219,7 @@ public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
         /* We destroy all bookmarks - even permanent ones when refreshing */
         valid = false;
         item_map_lock.@lock ();
-        SidebarItemInterface.item_id_map.unset (id);
+        item_id_map.unset (id);
         item_map_lock.unlock ();
         base.destroy ();
     }
@@ -557,7 +557,7 @@ public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
 
     private bool process_dropped_row (string drop_text, bool dropped_between) {
         var id = (uint32)(uint.parse (drop_text));
-        var item = SidebarItemInterface.get_item (id);
+        var item = get_item (id);
 
         if (item == null ||
             !dropped_between ||

--- a/src/View/Sidebar/BookmarkRow.vala
+++ b/src/View/Sidebar/BookmarkRow.vala
@@ -77,12 +77,12 @@ public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
         }
     }
 
-    public SidebarListInterface list { get; construct; }
-    public uint32 id { get; construct; }
+    public SidebarListInterface list { get; set construct; }
+    public uint32 id { get; set construct; }
     public string uri { get; set construct; }
     public Icon gicon { get; set construct; }
-    public bool pinned { get; construct; } // Cannot be dragged
-    public bool permanent { get; construct; } // Cannot be removed
+    public bool pinned { get; set construct; } // Cannot be dragged
+    public bool permanent { get; set construct; } // Cannot be removed
     public bool can_insert_before { get; set; default = true; }
     public bool can_insert_after { get; set; default = true; }
 

--- a/src/View/Sidebar/BookmarkRow.vala
+++ b/src/View/Sidebar/BookmarkRow.vala
@@ -179,7 +179,7 @@ public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
             propagation_phase = BUBBLE,
             button = 0
         };
-        button_controller.released.connect (button_release_event);
+        button_controller.released.connect (on_button_release_event);
 
         notify["gicon"].connect (() => {
             icon.set_from_gicon (gicon, Gtk.IconSize.MENU);
@@ -245,7 +245,7 @@ public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
         return false;
     }
 
-    protected virtual void button_release_event (int n_press, double x, double y) {
+    protected virtual void on_button_release_event (int n_press, double x, double y) {
         if (!valid) { //Ignore if in the process of being removed
             return;
         }

--- a/src/View/Sidebar/DeviceListBox.vala
+++ b/src/View/Sidebar/DeviceListBox.vala
@@ -21,8 +21,8 @@
  */
 
 public class Sidebar.DeviceListBox : Gtk.Box, Sidebar.SidebarListInterface {
-    public Files.SidebarInterface sidebar { get; construct; }
-    public Gtk.ListBox list_box { get; internal set; }
+    public Files.SidebarInterface sidebar { get; set construct; }
+    public Gtk.ListBox list_box { get; set construct; }
 
     private VolumeMonitor volume_monitor;
 

--- a/src/View/Sidebar/NetworkListBox.vala
+++ b/src/View/Sidebar/NetworkListBox.vala
@@ -21,8 +21,8 @@
  */
 
 public class Sidebar.NetworkListBox : Gtk.Box, Sidebar.SidebarListInterface {
-    public Files.SidebarInterface sidebar { get; construct; }
-    public Gtk.ListBox list_box { get; internal set; }
+    public Files.SidebarInterface sidebar { get; set construct; }
+    public Gtk.ListBox list_box { get; set construct; }
 
     public NetworkListBox (Files.SidebarInterface sidebar) {
         Object (sidebar: sidebar);

--- a/src/View/Window.vala
+++ b/src/View/Window.vala
@@ -807,7 +807,7 @@ public class Files.View.Window : Hdy.ApplicationWindow {
 
     private void add_window (GLib.File location = default_location, ViewMode mode = default_mode) {
         var new_window = new Window (marlin_app);
-        new_window.add_tab (location, real_mode (mode), false);
+        new_window.add_tab.begin (location, real_mode (mode), false);
         new_window.present ();
     }
 

--- a/src/View/Window.vala
+++ b/src/View/Window.vala
@@ -298,6 +298,10 @@ public class Files.View.Window : Hdy.ApplicationWindow {
                 case ViewMode.MILLER_COLUMNS:
                     app_menu.on_zoom_setting_changed (Files.column_view_settings, "zoom-level");
                     break;
+                case ViewMode.PREFERRED:
+                case ViewMode.CURRENT:
+                case ViewMode.INVALID:
+                    assert_not_reached (); //The switcher should not generate these modes
             }
         });
 

